pymess/backend/push/onesignal.py

```diff
@@ -0,0 +1,48 @@
+from http import HTTPStatus
+from json.decoder import JSONDecodeError
+
+import requests
+from django.utils import timezone
+from django.utils.encoding import force_text
+from onesignal import DeviceNotification, OneSignal
+
+from pymess.backend.push import PushNotificationBackend
+from pymess.config import settings
+from pymess.models import PushNotificationMessage
+from pymess.utils.logged_requests import generate_session
+
+
+class OneSignalPushNotificationBackend(PushNotificationBackend):
+
+    def _is_result_partial_error(self, result):
+        return not result.is_error and result.errors
+
+    def _is_invalid_result(self, result):
+        return result.is_error or self._is_result_partial_error(result)
+
+    def publish_message(self, message):
+        onesignal_client = OneSignal(settings.PUSH_NOTIFICATION_ONESIGNAL_APP_ID,
+                                     settings.PUSH_NOTIFICATION_ONESIGNAL_API_KEY)
+        onesignal_client.session = generate_session(slug='pymess - OneSignal', related_objects=(message,))
+
+        notification = DeviceNotification(
+            include_external_user_ids=(message.recipient,),
+            contents={'en': message.content},
```

Comment:
to se předpokládá anglický jazyk, nebo proč je to 'en'?

Reply:
OneSignal požaduje obsah zprávy v angličtině.